### ISSUE TITLE
sklearn.cross_validation being deprecated

### ISSUE THREAD

**Issue by bthirion** [nilearn/nilearn#1631]
It is used quite a lot in examples. We hsould update to sklearn.model_selection

**Comment by KamalakerDadi** [comment]
Yes indeed lot of examples uses cross_validation. That is the reason I hesitated to change them not to break compatibility. But, I think it will be the right moment to change.

**Comment by kchawla-pi** [comment]
The `__all__` in `cross_validation.py` and `model_selection/__init__.py` has several common public functions/classes (I am calling them _public imports_ here).
To make the changes I must change the examples to use the equivalent public imports.

It seems to me, where the public imports used are the same, all I have to do in the examples & docs is replace `sklearn.cross_validation.<function>` with `sklearn.model_selection.<same function>` .

The common public imports are:
```
KFold
cross_val_score  # Has an additional optional `groups` param in model_selection.
train_test_split
StratifiedKFold
check_cv
```
The non-common public import is:
```
cross_validation.LeaveOneLabelOut
```
That one has many mentions in the docs. What should we do about that? I will change the common ones first and then take a look at that one.

**Comment by kchawla-pi** [comment]
On further discovery, a bunch of functions have been renamed. `cross_validation.LeaveOneLabelOut` is now `model_selection.LeaveOneGroupOut` . I will dig in for more documentation regarding this and then lay it out.
http://scikit-learn.org/stable/whats_new.html in the section **Rename of splitter classes which accepts group labels along with data**

**Comment by KamalakerDadi** [comment]
Sounds good to me. We need to make users understand that they should use ```cross_validation``` or ```model_selection``` depending on their version. I think this can be done by adding a comment as note or better in a better way.

Good catch on the LeaveOneLabelOut thing. Looking forward for a PR on this issue.

**Comment by kchawla-pi** [comment]
I have searched the code and documentation using the website's search function and browser & IDE's find feature. I have listed 18 locations where modifications must be made. Some locations have multiple entries. We must ensure that the change does not break the functionality of the imported entities in the code.

Total 5 functional entities have been imported and used. 4 of them are common to cross_validation & model_selection, 1 of them is not.

Common:

    cross_val_score
    KFold
    StratifiedKFold
    train_test_split

Not-common:    

    LeaveOneLabelOut

I will start with the common imports and change them one at a time, then run tests.
